### PR TITLE
Add real-debrid-manager extension

### PR DESCRIPTION
**Pull request by YamenSharaf**
## **New extension**:  Real-Debrid Manager
A Real-Debrid client for Raycast.

## Screencast

https://github.com/raycast/extensions/assets/25423453/d38e71fc-388b-4e8d-8928-9d25bd240977

https://github.com/raycast/extensions/assets/25423453/49ec95e3-9a74-4c6f-bce2-4c74615818fc

### Downloads

- View and search Downloads.
- Unrestrict hoster links for [supported hosters](https://real-debrid.com/compare).
- Add links from clipboard.
- Copy direct download links.
- Copy original hoster file links.
- Open media files with installed media players ([IINA](https://iina.io/)/ [VLC](https://www.videolan.org/vlc/download-macosx.html) recommended).
- Delete downloads.

### Torrents

- View and search Torrents.
- Add new torrent via magnet links.
- Add new torrent via torrent file upload.
- Add magnet links from clipboard.
- Select files for new torrents and reselect for existing torrents.
- View files and sizes in uploaded torrent.
- Send torrent files to Downloads.
- Delete torrents.

### User Info

- Display user info and premium status.

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Review by mil3na — CHANGES_REQUESTED** [review]
Hi! Thank you so much for your contribution to Raycast!
My name is Milena and I will be reviewing your extension today :)

Other than the things I pointed out on the code, I have a couple more comments:

CleanShot 2023-07-14 at 14 02 56@2x

Error messages are not very friendly. I suggest that they should be mapped to real copy, instead of just spilling out what the API gives you.

Is there a way to have a RealDebrid URL without needing to subscribe? I'd like to test the download feature, but I've only got invalid URLs.

File: extensions/real-debrid-manager/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- Send torrent files to Downloads.
+- Delete torrents.
+
+### User Info
+
+- Display user info and premium status.
+
+# Further development
+
+Some ideas for further improvements:
```
[the remark this answers is not in the document]
Comment (YamenSharaf) [reply extensions/real-debrid-manager/README.md]: Removed! Probably no need for the announcement

File: extensions/real-debrid-manager/package.json [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      "icon": "command-icon.png"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "api_token",
+      "type": "password",
+      "required": true,
+      "title": "Real-Debrid Token",
+      "description": "You can find your token at https://real-debrid.com/apitoken",
```
[the remark this answers is not in the document]
Comment (YamenSharaf) [reply extensions/real-debrid-manager/package.json]: Removed

**Comment by YamenSharaf** [comment]
> Hi! Thank you so much for your contribution to Raycast! My name is Milena and I will be reviewing your extension today :)
> 
> Other than the things I pointed out on the code, I have a couple more comments:
> 
> CleanShot 2023-07-14 at 14 02 56@2x
> 
> Error messages are not very friendly. I suggest that they should be mapped to real copy, instead of just spilling out what the API gives you.

I've worked on cleaning up error messages so they don't show duplicate `Error` for example. Unfortunately, I don't have access to all possible errors from real-debrid's side and would like to give an indication to what exactly went wrong.

> 
> Is there a way to have a RealDebrid URL without needing to subscribe? I'd like to test the download feature, but I've only got invalid URLs.

Unfortunately, all of RD's functionality is gated behind a paid subscription. I could take a screen recording with more functionality if needed.
Just to clarify, the Downloads section doesn't download files to the disk, but these download items are actually listed from a remote server. The idea is that a user can open one of these direct links in the browser to initiate a download or send them to a locally installed media player to stream them directly.



Finally. I'm so sorry about the bulk of new changes since the review. I've been working over the past couple of days on some quality-of-life improvements as well as some refactoring. I wasn't aware that a review process had started when I pushed the changes 🙏🏻

**Comment by mil3na** [comment]
Hi!
In general, it looks good to me. Do you mind sharing a video of a common workflow for using this extension? Showing the QoL changes, if possible. As I don't have access to the service, that would be very helpful!

**Comment by YamenSharaf** [comment]
Thanks @mil3na for the review. RD is a service that makes it easy to download files from torrents or from hoster links like `mega.nz`or even YouTube. It uploads the content to a remote server and from there the download process becomes painless.

## Examples
### [Torrent file OR magnet](https://share.cleanshot.com/2GSMdzhQ)
- A user selects a torrent file or submits a magnet link
- They see a screen to confirm which file(s) from within the torrent they want to store on the server.
- After confirmation, the torrent is added to the torrent list where it's downloaded remotely (sometimes it's already cached d at a RD server).
- The list of torrents can be viewed. When complete, the selected torrent can be added to the download list to be downloaded and/or streamed.

### [Hoster or YouTube Link](https://share.cleanshot.com/qSzw6FgY)
- A user adds a YouTube Link from clipboard or the main add link commmand
- The link is added to the Downloads directly. Where it can be selected and information about it can be viewed.
- User has the choice to
  - Open in browser (download)
  - OR Stream in a local video player
  - OR stream in a web player made by RD
  - OR copy a direct download link

### [User can delete link or torrent from the list](https://share.cleanshot.com/Fmqnx88p)

## The improvements that were recently added
- Move the information of the torrent/download item to the metadata section as conventional
- Add new icons to the right-hand side of the list to indicate torrent status for torrents and file type for downloads
Screenshot 17 07 2023-8 42 52
Screenshot 17 07 2023-8 42 26

- Special handling for YouTUbe links where the thumbnail for the video is rendered 
Screenshot 17 07 2023-8 44 27

- Special handling for TV shows and movies where some more information is fetched and rendered
Screenshot 17 07 2023-8 47 00

**Review by mil3na — APPROVED** [review]
Thanks for your effort and the detailed explanation about your extension! LGTM!

**Pull request merged**